src/DatabaseSchemaManager.php

```diff
@@ -0,0 +1,251 @@
+<?php
```

Comment:
The fact that most method here require the `connection` as a parameter is a red flag... But I can't quite put my finger on how to better organize this....

Reply:
Well, I've solved that by creating another class `DatabaseSchema` and using the `DatabaseSchemaManager` inside it. (So the manager is the singleton and schema was the class used by us/developers). 
But then I had to duplicate alot of methods in both classes (duplicate method signatures, not method contents). 

So user would do: `DatabaseSchema::isColumnNullable()` that would translate to `$this->schemaManager->isColumnNullable()`. 

I reverted that change as I though for such a small thing that is not goind to expand further, it was overkill and would make it more difficult to understande where stuff come from.

The reason is because the $schema should persist but the connection is dynamic, depending on the caller so we can't rely on `$this->connection`, unless we create another class (not singleton) like I described above.